lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -0,0 +1,192 @@
+        address[:city]        = options[:billing_address][:city]
+        address[:state]       = options[:billing_address][:state]
+        address[:country]     = options[:billing_address][:country]
+        address[:line_1]      = options[:billing_address][:address1]
+        address[:postal_code] = options[:billing_address][:zip]
+      end
+
+      def add_invoice(post, money, options)
+        post[:checkout_reference] = options[:checkout_reference]
+        post[:amount]             = amount(money)
+        post[:currency]           = (options[:currency] || currency(money))
+        post[:description]        = options[:description]
+      end
+
+      def add_payment(post, payment, options)
+        post[:payment_type] = options[:payment_type] if PAYMENT_TYPES.include?(options[:payment_type])
+        card                = post[:card] = {}
+        card[:name]         = [payment.first_name, payment.last_name].join(' ')
+        card[:number]       = payment.number
+        card[:expiry_month] = format(payment.month, :two_digits)
+        card[:expiry_year]  = payment.year
+        card[:cvv]          = payment.verification_value
+      end
+
+      def commit(action, post, method = :post)
+        response = trigger_request(action, post.compact, method).symbolize_keys
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response),
+          test: test?,
+          error_code: error_code_from(response)
+        )
+      end
+
+      def trigger_request(action, post, method)
+        begin
+          response = parse(ssl_request(method, live_url + action, post.to_json, auth_headers))
+        rescue ResponseError => e
+          response = JSON.parse(e.response.body)
+        end
+
+        # Multiple invalid parameters
+        response = format_multiple_errors(response) if response.is_a?(Array)
+
+        return response
+      end
+
+      def parse(body)
+        JSON.parse(body)
+      end
+
+      def success_from(response)
+        response[:status] == 'PENDING'
+      end
+
+      def message_from(response)
+        return response[:status] if success_from(response)
+
+        response[:message] || response[:error_message]
+      end
+
+      def authorization_from(response)
+        response[:id]
+      end
+
+      def auth_headers
+        {
+          'Content-Type' => 'application/json',
+          'Authorization' => "Bearer #{options[:token]}"
+        }
+      end
+
+      def error_code_from(response)
+        response[:error_code] unless success_from(response)
+      end
+
+      def format_multiple_errors(responses)
+        errors = responses.map do |response|
+          { error_code: response['error_code'], param: response['param'] }
+        end
+
+        {
+          error_code: STANDARD_ERROR_CODE_MAPPING[:multiple_invalid_parameters],
+          message: 'Validation error',
```

Comment:
❓ Question:

Any reason to set here explicitly that all errors are going to be validation errors?

Reply:
Yes, in the [SumUp documentation](https://developer.sumup.com/docs/api/process-a-checkout/) the error 400 (Bad Request) have a response for Multiple_Invalid_Parameters all of them are `Vaidation error`.

Reply:
I mean => let's say that you are going to have 401 or 500 errors are they to be shown as "validation errors" any way ?

Reply:
No, according to the documentation there is only one type of Multiple_Invalid_Parameters (Array) so the scope is only for these errors (400 Bad Request)